workflow/graph.go

```diff
@@ -0,0 +1,178 @@
+package workflow
```

Comment:
This is basically a file renaming from algorithm.go to graph.go but with a new graph struct